api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -158,4 +176,12 @@ message CommonTlsContext {
   //
   // There is no default for this parameter. If empty, Envoy will not expose ALPN.
   repeated string alpn_protocols = 4;
+
+  // Config for fetching TLS certificates via CA client plugin.
+  // [#not-implemented-hide:]
+  CAClientPluginConfig tls_certificate_ca_client_plugin = 9;
+
+  // Config for fetching validation context via CA client plugin.
+  // [#not-implemented-hide:]
+  CAClientPluginConfig validation_context_ca_client_plugin = 10;
```

Comment:
Should this be inside `oneof validation_context_type` and be mutually exclusive with `SdsSecretConfig validation_context_sds_secret_config = 7;`?

Reply:
Good catch. Yes it should be moved under `oneof validation_context_type`. Once moved there it will be mutually exclusive with all the other members in there not just `validation_context_sds_secret_config`.  I assume there is no backward compatibility issue with adding a new member to a "oneof"

Reply:
Cool. And maybe the one above (`CAClientPluginConfig tls_certificate_ca_client_plugin = 9`) also needs to be mutually exclusive with `repeated SdsSecretConfig tls_certificate_sds_secret_configs = 6`. I'm not very sure though since `tls_certificate_sds_secret_configs` is a repeated field. Maybe Envoy meant to support fetching tls certificate from multiple sources; in that case the current way is probably ok as it just adds one more source to the list.

Reply:
The main problem here is backward compatibility: moving an existing field (`tls_certificate_sds_secret_configs`) to oneof creates backward compatibility issues so we decided against it. However this might be a good candidate for the field_migrate annotations. @htuch  do you agree?

Reply:
Just use the oneof promotion annotation for that, document in the interim.